evennia/commands/default/building.py

```diff
@@ -1515,31 +1518,130 @@ def check_attr(self, obj, attr_name):
         """
-        if obj.attributes.has(attr):
-            return "\nAttribute %s/%s = %s" % (obj.name, attr,
-                                               obj.attributes.get(attr))
+        for key, nested_keys in self.split_nested_attr(attr):
+            if obj.attributes.has(key):
+                val = obj.attributes.get(key)
+                val = self.do_nested_lookup(val, *nested_keys)
+                if val is not self.not_found:
+                    return "\nAttribute %s/%s = %s" % (obj.name, attr, val)
         else:
             return "\n%s has no attribute '%s'." % (obj.name, attr)
 
     def rm_attr(self, obj, attr):
         """
-        Remove an attribute from the object, and report back.
+        Remove an attribute from the object, or a nested data structure, and report back.
         """
-        if obj.attributes.has(attr):
-            val = obj.attributes.has(attr)
-            obj.attributes.remove(attr)
-            return "\nDeleted attribute '%s' (= %s) from %s." % (attr, val, obj.name)
-        else:
-            return "\n%s has no attribute '%s'." % (obj.name, attr)
+        for key, nested_keys in self.split_nested_attr(attr):
+            if obj.attributes.has(key):
+                if nested_keys:
+                    del_key = nested_keys[-1]
+                    val = obj.attributes.get(key)
+                    deep = self.do_nested_lookup(val, *nested_keys[:-1])
+                    if deep is not self.not_found:
+                        try:
+                            del deep[del_key]
+                        except (IndexError, KeyError, TypeError):
+                            continue
+                    return "\nDeleted attribute '%s' (= nested) from %s." % (attr, obj.name)
+                else:
+                    exists = obj.attributes.has(key)
+                    obj.attributes.remove(attr)
+                    return "\nDeleted attribute '%s' (= %s) from %s." % (attr, exists, obj.name)
+        return "\n%s has no attribute '%s'." % (obj.name, attr)
 
     def set_attr(self, obj, attr, value):
+        done = False
+        for key, nested_keys in self.split_nested_attr(attr):
+            if obj.attributes.has(key) and nested_keys:
+                acc_key = nested_keys[-1]
+                lookup_value = obj.attributes.get(key)
+                deep = self.do_nested_lookup(lookup_value, *nested_keys[:-1])
+                if deep is not self.not_found:
+                    # To support appending and inserting to lists
+                    # a key that starts with @ will insert a new item at that
+                    # location, and move the other elements down.
+                    # Just '@' will append to the list
+                    if isinstance(acc_key, str) and acc_key[0] == LIST_APPEND_CHAR:
+                        try:
+                            if len(acc_key) > 1:
+                                where = int(acc_key[1:])
```

Comment:
If the user accidentally enters something wrong, like trying (wrongly) to use `[+'key']`, maybe with the intent of "appending" to a dictionary (which is an easy mistake to make if you are not too Python savvy), this will lead to a traceback about not being able to convert to an int. Best to catch the `ValueError` here as well and to give a proper error message rather than the raw traceback.

Reply:
Nice catch! I'll add a test :)